wolfcrypt/src/cmac.c

```diff
@@ -237,6 +237,10 @@ int wc_CmacFinal(Cmac* cmac, byte* out, word32* outSz)
         subKey = cmac->k1;
     }
     else {
+        /* ensure we will have a valid remainder value */
```

Comment:
This must be below the `word32 remainder`, so this section needs to be re-written like:

```
word32 remainder;
if (cmac->bufferSz > AES_BLOCK_SIZE) {
    return BAD_STATE_E;
}
remainder = AES_BLOCK_SIZE - cmac->bufferSz;
```

Reply:
yes. good catch. updated.